vector/src/main/java/im/vector/app/features/invite/InviteUsersToRoomViewModel.kt

```diff
@@ -63,11 +65,9 @@ class InviteUsersToRoomViewModel @AssistedInject constructor(
                             is PendingSelection.UserPendingSelection     -> room.membershipService().invite(user.user.userId, null)
                             is PendingSelection.ThreePidPendingSelection -> room.membershipService().invite3pid(user.threePid)
                         }
-                    }
-                    .catch { cause ->
-                        _viewEvents.post(InviteUsersToRoomViewEvents.Failure(cause))
-                    }
-                    .collect {
+                    }.onCompletion { error ->
```

Comment:
It seems like ordering here is very important. If we use `catch` before `onCompletion`,  the possible error value that `onCompletion` can receive will always be null and we won't be able to distinguish between successful and unsuccessful results.

Reply:
Good to know, thanks!